examples/vue-3-cli/src/stories/Button.stories.js

```diff
@@ -0,0 +1,39 @@
+import MyButton from './Button.vue';
+
+export default {
+  title: 'Example/Button',
+  component: MyButton,
+  argTypes: {
+    backgroundColor: { control: 'color' },
+    size: { control: { type: 'select', options: ['small', 'medium', 'large'] } },
+  },
+};
+
+const Template = (args, { argTypes }) => ({
+  props: Object.keys(argTypes),
+  components: { MyButton },
+  template: '<my-button v-bind="$props" />',
+});
```

Comment:
This might not be the right time to ask this, as I understand it works the same way in vue2, but am I right in saying that the package automatically maps `context.args` (the first argument) to `$props` in the component?

Is there some way to override this behaviour in the story/template definition? I'm wondering if we could support something like:

```js
const Template = (args, { argTypes }) => ({
    props: [...Object.keys(argTypes), 'extraArg'],
    components: { MyButton },
    template: '<my-button v-bind="$props" />',
    args: {
      ...args,
      extraArg: 'value',
    },
  });
```

cc @jonniebigodes -- this came up in the development of the vue snippets, IIRC.

Reply:
I honestly have no idea. What happens is that the `render` function receives `args` and those are set in shallowRef as the `props`. Those are then provided to components as `props` (2nd argument to `h(Component, props)`) and also use a `provide()` function to be injected as `props` too.

Reply:
@tmeasday something like this in vue3 (untested, based on discussion with @phated) would be the vue way of doing things:

```js
const Template = (args, { argTypes }) => ({
  props: [...Object.keys(argsTypes), 'extraArg'],
  components: { MyButton },
  setup(props) {
    return () => h(MyButton, { ...props, extraArg: 'foo' })
  }
});
```

We could also potentially bastardize things to make args first class in Vue stories, but this could be confusing to Vue developers since args don't exist in the Vue world. That might look something like this:

```js
const Template = (args) => ({
  components: { MyButton },
  template: '<my-button v-bind="$props" />',
  args: { ...args, extraArg: 'value' },
});
```

We'd need to post-process this in the Vue framework code to turn it into a valid Vue component.

Reply:
So the issue here is the "thing" returned by a vue story is a real existing thing in Vue (essentially equivalent to a component in react), so adding an extra `args` property would be sort of bastardizing it. I get that.

I think there's sort of a mismatch here between the semantics of a vue and a react story.

```js
// In React
const Template = (args) => <Component {...args}>;

// this means:
const Template = /* a "story component" implemented as a SFC */

// In Vue
const Template = (args) = ({ ... });

// this means
const Template = (args) => /* a "story component" */
```

Notice the difference? In Vue we are returning a component from the function, which will later be "instanciated" (with the args) via the vue framework. That's like fundamentally different, which is why things are a bit hairy.

I suspect (based on not much knowledge) that really the function could be dropped from the vue template. I wonder if that would make things more consistent, and it more natural to do things like modify the args.